### PR TITLE
Added Backward Compatibility for Environment Application in Project Environment

### PR DESCRIPTION
**Pull request by manas-droid**
Thank you for your contribution.
Before submitting this PR, please make sure:

- [ ] PR description and commit message should describe the changes done in this PR
- [ ] Verify the PR is pointing to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [ ] Latest Code from master or specific release branch is merged to your branch
- [ ] No unwanted\commented\junk code is included
- [ ] No new warning upon build solution
- [ ] Code Summary\Comments are added to my code which explains what my code is doing
- [ ] Existing unit test cases are passed
- [ ] New Unit tests are added for your development
- [ ] Sanity Tests are successfully executed for New and Existing Functionality
- [ ] Verify that changes are compatible with all relevant browsers and platforms.
- [ ] After creating pull request there should not be any conflicts
- [ ] Resolve all Codacy comments
- [ ] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments
- [ ] Update the Help Library document to match any feature changes

## Summary by CodeRabbit

- **New Features**
	- Enhanced the environment setup wizard to include application platform selection and display corresponding icons.
	- Introduced logic to handle application platforms and selection in the `AddApp` method.
- **Improvements**
	- Updated the internal handling of application platform types to improve user experience in environment configuration.
	- Added new icons for better visual representation of different environments in the project resources.
- **Bug Fixes**
	- Fixed a message formatting issue in user notifications for better clarity.

### CODE REVIEW

File: Ginger/Ginger/Environments/AppsListPage.xaml.cs [hunk 1]
```diff
@@ -36,50 +39,65 @@ namespace Ginger.Environments
...
         {
-            EnvApplication app = new EnvApplication();
-            app.Name = "New";
-            app.Active = true;
-            AppEnvironmnet.Applications.Add(app);
+            var ApplicationPlatforms = WorkSpace.Instance.Solution.ApplicationPlatforms.Where((app) => !AppEnvironment.CheckIfApplicationPlatformExists(app.Guid, app.AppName))?.ToList();
+
+
+
+            if (ApplicationPlatforms == null || ApplicationPlatforms?.Count == 0)
```
Comment (Maheshkale447) [inline Ginger/Ginger/Environments/AppsListPage.xaml.cs:72]: ApplicationPlatforms?.Count == 0 keep only this condition as ? will be already checking the null condition

File: Ginger/GingerCoreCommon/ReporterLib/UserMsgsPool.cs [hunk 2]
```diff
@@ -247,7 +247,8 @@ public static void LoadUserMsgsPool()
...
             Reporter.UserMsgsPool.Add(eUserMsgKey.RunSetLoadFromReportError, new UserMsg(eUserMsgType.ERROR, "Unable To Load RunSet", "{0}", eUserMsgOption.OK, eUserMsgSelection.OK));
             Reporter.UserMsgsPool.Add(eUserMsgKey.PublishApplicationToOtherEnv, new UserMsg(eUserMsgType.QUESTION, "Publish Application to other Environment", "Do you want to publish this application to all the other existing environments?", eUserMsgOption.YesNo, eUserMsgSelection.No));
-            Reporter.UserMsgsPool.Add(eUserMsgKey.NoApplicationPlatformLeft, new UserMsg(eUserMsgType.INFO, "No Application is available to add", "All the applications are already available in {0} Environment. Please add new application in the Configurations-> Target application page to add application here", eUserMsgOption.OK, eUserMsgSelection.None));
+
+            Reporter.UserMsgsPool.Add(eUserMsgKey.NoApplicationPlatformLeft, new UserMsg(eUserMsgType.INFO, "No Application is available to add", "All the applications are already available in {0} Environment. \nPlease add new application in the Configurations-> Target application page to add application here", eUserMsgOption.OK, eUserMsgSelection.None));
```
Comment (Maheshkale447) [inline Ginger/GingerCoreCommon/ReporterLib/UserMsgsPool.cs:251]: Use the Ginger resource dictionary instead of Target Application

**Review by Maheshkale447 — CHANGES_REQUESTED** [review]

File: Ginger/Ginger/Environments/AppsListPage.xaml.cs (hunk 1 above)
Comment (manas-droid) [reply Ginger/Ginger/Environments/AppsListPage.xaml.cs]: Ok done

File: Ginger/GingerCoreCommon/ReporterLib/UserMsgsPool.cs (hunk 2 above)
Comment (manas-droid) [reply Ginger/GingerCoreCommon/ReporterLib/UserMsgsPool.cs]: ok Implemented the requested changes

**Review by Maheshkale447 — APPROVED** [review]

**Pull request merged**